mathics/core/expression.py

```diff
@@ -1837,24 +1837,29 @@ def thread(self, evaluation, head=None) -> typing.Tuple[bool, "Expression"]:
             leaves = [Expression(self._head, *item) for item in items]
             return True, Expression(head, *leaves)
 
-    def is_numeric(self) -> bool:
-        return (
-            self._head.get_name()
-            in system_symbols(
-                "Sqrt",
-                "Times",
-                "Plus",
-                "Subtract",
-                "Minus",
-                "Power",
-                "Abs",
-                "Divide",
-                "Sin",
+    def is_numeric(self, evaluation=None) -> bool:
+        if evaluation:
+            if not "System`NumericFunction" in evaluation.definitions.get_attributes(
+                self._head.get_name()
+            ):
+                return False
+            return all(leaf.is_numeric(evaluation) for leaf in self._leaves)
+        else:
+            return (
+                self._head.get_name()
+                in system_symbols(
```

Comment:
I am surprised that we don't have a way to tag functions that return numeric values. 

This kind of code seems fragile and is probably not complete.

Reply:
Also it might be good to compute this once such as in a class variable rather than each time is_numeric is called.